tinyexpr.c

```diff
@@ -113,6 +119,20 @@ void te_free(te_expr *n) {
 
 static double pi() {return 3.14159265358979323846;}
 static double e() {return 2.71828182845904523536;}
+static double fac(double a) {/* simplest version of fac */
+    if (a < 0.0)
+        return NAN;
+    if (a > UINT_MAX)
+        return INFINITY;
+    unsigned int ua = (unsigned int)(a);
+    unsigned long int result = 1, i = 1;
+    for (i = 1; i <= ua; i++) {
```

Comment:
That much I managed to gather, this is a nitpick, you have one assignment in the declaration and one in inside the `for` initialization.